move view footer control state to redux

uladzimirdev opened the pull request:
refactor state handling in view footer control - rely on global state instead of multiple useEffect

❌ two failing tests are expected

### how to verify
- try using back button in the browser
- try switching between notebook editor and visualization
- try reloading the page at any state
- try to convert question to sql

### demo




https://github.com/user-attachments/assets/35980d8e-bfb8-460a-9722-5b2e7cc52b91

uladzimirdev committed:
move view footer control state to redux

uladzimirdev committed:
fix typecheck

uladzimirdev committed:
Merge branch 'qb-predictable-sticky-buttons' into qb-predictable-editor-state-redux

ranquild reviewed (commented):
1 comment, looks good

uladzimirdev commented on frontend/src/metabase/query_builder/components/view/ViewFooter/ViewFooterControl.tsx:
```diff
@@ -36,55 +37,30 @@ const ViewFooterControl = ({
-      setValue("visualization");
-    }
-  }, [isNative, isNotebook, value]);
+  const initialValue = isNotebook
```
initializeQB is called only once, but we need to listen to back/forth browser buttons

https://github.com/user-attachments/assets/ebd46ae8-9d72-4a07-a39e-fee13a815f2e

Do you have in mind another place where we can initialize the value? otherwise, I'd keep this logic in the component

uladzimirdev replied:
fixed [94e1dcc](https://github.com/metabase/metabase/pull/50498/commits/94e1dccb4797ff147a4e26b74e5e478746e5b127)

uladzimirdev committed:
fix logic

romeovs commented:
Not sure if any of this is intended behavior but I'm noticing some unexpected things when using the back button or refreshing.

1. The state of the footer control does not change to vizualisation when clicking the back button.
  
https://github.com/user-attachments/assets/a6e32687-6ab4-4c26-b9ba-a68ecbff0dc2

2. The current tab gets lost when refreshing:

https://github.com/user-attachments/assets/49e4b447-8d79-4971-b1ce-60838425a904

3. (a but of a pedantic one) in some cases the visualization gets lost on refresh

https://github.com/user-attachments/assets/5dd7615e-6811-4859-8cce-c3ab5092f725

uladzimirdev commented:
nice catch @romeovs, indeed first click on back button doesn't work

uladzimirdev committed:
move logic to redux

uladzimirdev commented:
@romeovs I addressed comment 1. as for comment 2, I think you uploaded the same video, but I got the idea: if you choose "results" and refresh the page, "visualization" will be selected - it's fine as for me and it follows current logic.

as for 3, it's out of current scope

uladzimirdev committed:
Merge branch 'qb-predictable-sticky-buttons' into qb-predictable-editor-state-redux

uladzimirdev committed:
fix new test
